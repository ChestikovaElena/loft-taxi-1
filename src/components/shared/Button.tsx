import React from 'react';
import styled from 'styled-components';

const Button = ({ name }: { name: string }) => <StyledButton type="submit">{name}</StyledButton>;

export default Button;

const StyledButton = styled.button`
    align-self: flex-end;
    padding: 15px 50px;
    font-size: 21px;
    font-weight: 400;
    border: 0;
    border-radius: 4px;
    background-color: #ffc617;
`;
