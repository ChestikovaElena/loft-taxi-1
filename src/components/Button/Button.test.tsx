import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Heading', () => {
    it('компонент рендерится без ошибок', () => {
        const button = shallow(<Button />);

        expect(button).toBeTruthy;
    });

    it('тест кнопки совпадает с текстом, переданным ей в пропсах', () => {
        const button = shallow(<Button name="Кнопка" />);

        expect(button.text()).toContain('Кнопка');
    });
});
