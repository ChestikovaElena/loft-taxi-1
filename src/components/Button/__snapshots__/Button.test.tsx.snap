// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button компонент отображается корректно 1`] = `
<button
  className="sc-bdVaJa cTCHOq"
  data-testid="auth-btn"
  type="submit"
/>
`;
